src/lucky/rate_limit.cr

```diff
@@ -0,0 +1,41 @@
+module Lucky::RateLimit
+  macro included
+    before enforce_rate_limit
+  end
+
+  abstract def rate_limit : NamedTuple(to: Int32, within: Time::Span)
+
+  private def enforce_rate_limit
+    cache = LuckyCache.settings.storage
+    count = cache.fetch(rate_limit_key, as: Int32, expires_in: rate_limit["within"]) { 0 }
+    cache.write(rate_limit_key, expires_in: rate_limit["within"]) { count + 1 }
+
+    if count > rate_limit["to"]
+      context.response.status = HTTP::Status::TOO_MANY_REQUESTS
+      context.response.headers["Retry-After"] = rate_limit["within"].to_s
+      plain_text("Rate limit exceeded")
+    else
+      continue
+    end
+  end
+
+  private def rate_limit_key : String
+    klass = self.class.to_s.downcase.gsub("::", ":")
+    "ratelimit:#{klass}:#{rate_limit_identifier}"
+  end
+
+  private def rate_limit_identifier : Socket::Address | Nil
```

Comment:
This code is a copy of the RemoteIP code. You can override the method in your action, but the default needs something and I think the IP makes sense. The specs pass in a HTTP::Request so I don't get `context.remote_ip`.

Reply:
I think you should have access to `request.remote_ip`  https://github.com/luckyframework/lucky/blob/bddb10b79c35699bd51bba91897f52d0114fab3b/src/charms/request_extensions.cr#L1-L8

This is actually patched in Lucky. `context` doesn't have a `remote_ip` method, but you may be thinking of `context.request_id` which is something different. 

Now, with that said, it brings up the point that if you don't have the `RemoteIpHandler` in your middleware stack, that could cause issues... I wonder if there's a way that we could say you must have that handler in your stack in order to use this module? :thinking: